scripts/fix_error_ids.py

```diff
@@ -102,38 +103,93 @@ def fix_ids(used_ids, file_names):
         fix_ids_in_file(file_name, available_ids, used_ids)
 
 
-def find_source_files(top_dir):
-    """Builds the list of .h and .cpp files in top_dir directory"""
+def find_files(top_dir, sub_dirs, extensions):
+    """Builds a list of files with given extensions in specified subdirectories"""
 
     source_file_names = []
-    dirs = ['libevmasm', 'liblangutil', 'libsolc', 'libsolidity', 'libsolutil', 'libyul', 'solc']
-
-    for dir in dirs:
+    for dir in sub_dirs:
         for root, _, file_names in os.walk(os.path.join(top_dir, dir), onerror=lambda e: exit(f"Walk error: {e}")):
             for file_name in file_names:
                 _, ext = path.splitext(file_name)
-                if ext in [".h", ".cpp"]:
+                if ext in extensions:
                     source_file_names.append(path.join(root, file_name))
 
     return source_file_names
 
 
+def find_ids_in_test_file(file_name, ids):
+    source = read_file(file_name)
+    for m in re.finditer(TEST_FILE_PATTERN, source, flags=re.MULTILINE):
+        id = m.group(0)[-5:-1]
+        ids.add(id)
+
+
+def find_ids_in_test_files(file_names):
+    used_ids = set()
+    for file_name in file_names:
+        find_ids_in_test_file(file_name, used_ids)
+    return used_ids
+
+
+def print_ids(ids):
+    for k, id in enumerate(sorted(ids)):
+        if k % 10 > 0:
+            print(" ", end="")
+        elif k > 0:
+            print()
+        print(id, end="")
+
+
+def examine_id_coverage(top_dir, used_ids):
+    test_file_names =  find_files(
+        top_dir,
+        [path.join("test", "libsolidity")],
```

Comment:
Can you check which ones we loose if we restrict this to `syntaxTests` only?

Reply:
In addition to `syntaxTests`, there are 6 error codes in `recoveryTests` and 41 in `smtCheckerTests`. I listed three directories explicitly.

Reply:
Ok, makes sense!